stackdriver_test.go

```diff
@@ -1091,3 +1094,84 @@ func diffCreateMsg(want, got *monitoringpb.CreateTimeSeriesRequest) string {
 	}
 	return out
 }
+
+func TestCustomMonitorResource(t *testing.T) {
+	checkLabels := func(sink *Sink, labels map[string]string) error {
+		expectedLabels := sink.monitoredResource.GetLabels()
+
+		expectedLabelsBytes, _ := json.Marshal(expectedLabels)
+		labelsBytes, _ := json.Marshal(labels)
+
+		if expectedLabelsBytes == nil || !bytes.Equal(expectedLabelsBytes, labelsBytes) {
```

Comment:
Instead of using json.Marshal and comparing outputs, use [cmp.Diff](https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff):

```
if diff := cmp.Diff(sink.monitoredResource.GetLabels(), labels); diff != "" {
  t.Errorf("Monitored Resource labels mismatch (-want +got):\n%s", diff)
}
```

More info on this is here: https://github.com/golang/go/wiki/TestComments#equality-comparison-and-diffs

I think if you make that change you'll be able to remove the checkLabels helper entirely.